spec/unit/bootstrap/bundler_spec.rb

```diff
@@ -88,8 +88,8 @@
 
     context 'abort with an exception' do
       it 'gem conflict' do
-        allow(::Bundler::CLI).to receive(:start).with(bundler_args) { raise ::Bundler::SolveFailure.new('conflict') }
-        expect { subject }.to raise_error(::Bundler::SolveFailure)
+        allow(::Bundler::CLI).to receive(:start).with(bundler_args) { raise ::Bundler::VersionConflict.new('conflict') }
```

Comment:
2 minor nits:

 - When stubbing behavior with constraints using `with`, I have a preference for using the fluent interface ([`and_return`](https://rspec.info/features/3-12/rspec-mocks/configuring-responses/returning-a-value/), [`and_raise`](https://rspec.info/features/3-12/rspec-mocks/configuring-responses/raising-an-error/), etc.) over the block implementation wherever possible
 - Since we _expect_ the CLI to receive this message with these arguments, using RSpec's `expect` will make it easier to catch the cause of failure if/when this spec breaks.

```suggestion
        expect(::Bundler::CLI).to receive(:start).with(bundler_args).and_raise( ::Bundler::VersionConflict, 'conflict')
```

Reply:
while I agree with the change, I want to constrain the changeset of this PR to the work needed to remove the custom bundler.
Something we could and should do is an audit to all misuses of allow/expect (e.g. common use of  `expect` in `before` to mock behavior). I'll open a PR just for this change

Reply:
https://github.com/elastic/logstash/pull/15095